packages/vite/src/node/config.ts

```diff
@@ -766,10 +766,10 @@ function mergeConfigRecursively(
 }
 
 export function mergeConfig(
-  defaults: Record<string, any>,
-  overrides: Record<string, any>,
+  defaults: UserConfig,
+  overrides: UserConfig,
   isRoot = true
-): Record<string, any> {
+): UserConfig {
```

Comment:
After looking at the implementation, I think we just need to limit the output?

```diff
+ ): UserConfig {
- ): Record<string, any> & UserConfig {
```

Reply:
Do you mean adding it to both arguments and return types or just the return types?

Reply:
AFAIK `UserConfig` seems to be a subtype of `Record<string, any>`, which implies that it should have no effect if we just add it with an intersection.

Reply:
ok. But I think the params does not need to be changed, if you want to have type hint in the params you can do like this. 

```ts
mergeConfig(defineConfig({
  ...
}), config)
```

As your parameters change, the return also changes. So I don't think this type of assertion is needed and what do you think?

Reply:
OK. Perhaps we need an issue for further discussion, then decide how to implement it.